port/unix/omrsl.c

```diff
@@ -438,3 +441,90 @@ omrsl_startup(struct OMRPortLibrary *portLibrary)
 {
 	return 0;
 }
+
+/**
+ * Native Libraries
+ *
+ * This function is called go get all the shared  libraries loaded by a process. 
+ *
+ * @param[in] portLibrary Pointer to the OMR port library.
+ * @param[in] callback Function to be called for each library found.
+ * @param[in] userData User-defined data passed to the callback.
+ * @param[out] function pointer to the function 
+ *
+ * @return 0 if successful, or the first non-zero return value from the callback.
+ */
+uintptr_t
+omrsl_get_libraries(struct OMRPortLibrary *portLibrary, OMRLibraryInfoCallback callback, void *userData)
+{
+#if defined(LINUX)
+	char buffer[READ_CHUNK_SIZE + 1];
+	char carryOver[READ_CHUNK_SIZE + 1];
+	intptr_t bytesRead = 0;
+	uintptr_t result = 0;
+	void*  addrLow = NULL;
+	void* addrHigh = NULL;
+	unsigned long offset = 0;
+	int devMajor = 0;
+	int devMinor = 0;
+	unsigned long inode = 0;
+	char permissions[5] = {0};
+	char path[READ_CHUNK_SIZE];
+	int count = 0 ;
+	char combinedBuffer[2 * READ_CHUNK_SIZE + 1];
+	intptr_t fd = 0;
+	int lastCharIsNewline = 0;
+	char* line = NULL;
+	char* nextLine=NULL;
+	fd=portLibrary->file_open(portLibrary,"/proc/self/maps",EsOpenRead,0);
+	if (0 > fd) {
+		return -1;
+	}
+	bytesRead = portLibrary->file_read(portLibrary, fd, buffer, READ_CHUNK_SIZE);
+	do  {
```

Comment:
Moving the read inside the loop avoids repetition:
```c
	do  {
		bytesRead = portLibrary->file_read(portLibrary, fd, buffer, sizeof(buffer) - 1);
```
(line 522 should then be deleted)